spec/migrations/20200126203714_remove_vim_types_from_ems_events_spec.rb

```diff
@@ -0,0 +1,48 @@
+require_migration
+
+describe RemoveVimTypesFromEmsEvents do
+  let(:event_stream_stub) { migration_stub(:EventStream) }
+
+  migration_context :up do
+    it "converts a VimHash to a normal Hash" do
+      full_data = <<~FULL_DATA
+      --- !ruby/hash-with-ivars:VimHash
+      elements:
+        key: !ruby/string:VimString
+          str: '585137'
+          xsiType: :SOAP::SOAPInt
+          vimType:
+        datacenter: !ruby/hash-with-ivars:VimHash
+          elements:
+            name: !ruby/string:VimString
+              str: dev-vc67-DC
+              xsiType: :SOAP::SOAPString
+              vimType:
+            datacenter: !ruby/string:VimString
+              str: datacenter-104
+              xsiType: :ManagedObjectReference
+              vimType: :Datacenter
+          ivars:
+            :@xsiType: :DatacenterEventArgument
+            :@vimType:
+        eventType: VmCreatedEvent
+      ivars:
+        :@xsiType: :VmCreatedEvent
+        :@vimType:
+      FULL_DATA
+
+      event = event_stream_stub.create!(:full_data => full_data, :source => "VC")
+
+      migrate
+
+      full_data = YAML.load(event.reload.full_data)
+
+      expect(full_data.class.name).to eq("Hash")
+      expect(full_data["key"].class.name).to eq("String")
```

Comment:
Should the value also be verified here?

Reply:
Also, `expect(full_data["key"].class.name).to eq("String")` can be `expect(full_data["key"]).to be_kind_of(String)`

Reply:
:+1: will do

Reply:
> Also, expect(full_data["key"].class.name).to eq("String") can be expect(full_data["key"]).to be_kind_of(String)

The problem with that is a `VimString < String` so `expect(VimString).to be_kind_of(String)` passes

Reply:
Alright pushed a change to check the values of key and datacenter

Reply:
`.class == String` then?

Reply:
Okay